### ISSUE TITLE
Companion is broken with latest supervisor

### ISSUE THREAD

**Issue by LaneaLucy** [ESPresense/ESPresense-companion#490]
[17:16:34 ERR] Failed to get MQTT config from Hass Supervisor
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'System.Net.Http.HttpConnectionResponseContent'.
   at System.Net.Http.HttpContent.ReadAsStringAsync(CancellationToken cancellationToken)
   at Flurl.Http.FlurlResponse.GetJsonAsync[T]()
   at ESPresense.Services.MqttCoordinator.GetClient() in /App/src/Services/MqttCoordinator.cs:line 53
[17:16:34 INF] Now listening on: http://[::]:8267
[17:16:34 INF] Application started. Press Ctrl+C to shut down.
[17:16:34 INF] Hosting environment: Production
[17:16:34 INF] Content root path: /App
[17:16:34 INF] Connecting to mqtt server at localhost:1883...
[17:16:34 INF] MQTT {"primary": true} disconnected
[17:16:34 ERR] MQTT {"primary": true} connection failed Connecting with MQTT server failed (NotAuthorized).: null

**Comment by DTTerastar** [comment]
Thank you for reporting this, i'll look into it.

**Comment by 51av0sh** [comment]
I can confirm the same. ESPresense was unable to fetch the mqtt details from home assistant but it will connect if you manually add your login details to the config file. However, the web UI doesn't load for me anymore. It's just a blank page. Restarted several times with no result. 

![Screenshot 2023-12-24 164130](https://github.com/ESPresense/ESPresense-companion/assets/85576728/93ca0a07-20e9-4eac-b7e3-4ccf69a06260)

**Comment by chrisgilldc** [comment]
Likewise confirming I'm seeing the same issue. This is without making any configuration changes. Web UI still loads, just doesn't show any devices.

```
[07:57:33 WRN] Storing keys in a directory '/root/.aspnet/DataProtection-Keys' that may not be persisted outside of the container. Protected data will be unavailable when container is destroyed. For more information go to https://aka.ms/aspnet/dataprotectionwarning
[07:57:33 INF] Optimization enabled
[07:57:33 WRN] No XML encryptor configured. Key {1d3d33f5-4b05-47fe-85ef-12958dcb7b62} may be persisted to storage in unencrypted form.
[07:57:33 WRN] Overriding HTTP_PORTS '8080' and HTTPS_PORTS ''. Binding to values defined by URLS instead 'http://+:8267'.
[07:57:33 INF] Now listening on: http://[::]:8267
[07:57:33 INF] Application started. Press Ctrl+C to shut down.
[07:57:33 INF] Hosting environment: Production
[07:57:33 INF] Content root path: /App
[07:57:33 ERR] Failed to get MQTT config from Hass Supervisor
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'System.Net.Http.HttpConnectionResponseContent'.
   at System.Net.Http.HttpContent.ReadAsStringAsync(CancellationToken cancellationToken)
   at Flurl.Http.FlurlResponse.GetJsonAsync[T]()
   at ESPresense.Services.MqttCoordinator.GetClient() in /App/src/Services/MqttCoordinator.cs:line 53
[07:57:33 INF] Connecting to mqtt server at localhost:1883...
[07:57:34 INF] MQTT {"primary": true} disconnected
[07:57:34 ERR] MQTT {"primary": true} connection failed Connecting with MQTT server failed (NotAuthorized).: null
```

**Comment by chrisgilldc** [comment]
I will note I restored a backup of version 0.5.2 which does not have this issue, so it's a regression between 0.5.2 and 0.6.0.

**Comment by arne182** [comment]
Adding the Mqtt user and password to the config made everything work

**Comment by 51av0sh** [comment]
> Likewise confirming I'm seeing the same issue. This is without making any configuration changes. Web UI still loads, just doesn't show any devices.
> 
> ```
> [07:57:33 WRN] Storing keys in a directory '/root/.aspnet/DataProtection-Keys' that may not be persisted outside of the container. Protected data will be unavailable when container is destroyed. For more information go to https://aka.ms/aspnet/dataprotectionwarning
> [07:57:33 INF] Optimization enabled
> [07:57:33 WRN] No XML encryptor configured. Key {1d3d33f5-4b05-47fe-85ef-12958dcb7b62} may be persisted to storage in unencrypted form.
> [07:57:33 WRN] Overriding HTTP_PORTS '8080' and HTTPS_PORTS ''. Binding to values defined by URLS instead 'http://+:8267'.
> [07:57:33 INF] Now listening on: http://[::]:8267
> [07:57:33 INF] Application started. Press Ctrl+C to shut down.
> [07:57:33 INF] Hosting environment: Production
> [07:57:33 INF] Content root path: /App
> [07:57:33 ERR] Failed to get MQTT config from Hass Supervisor
> System.ObjectDisposedException: Cannot access a disposed object.
> Object name: 'System.Net.Http.HttpConnectionResponseContent'.
>    at System.Net.Http.HttpContent.ReadAsStringAsync(CancellationToken cancellationToken)
>    at Flurl.Http.FlurlResponse.GetJsonAsync[T]()
>    at ESPresense.Services.MqttCoordinator.GetClient() in /App/src/Services/MqttCoordinator.cs:line 53
> [07:57:33 INF] Connecting to mqtt server at localhost:1883...
> [07:57:34 INF] MQTT {"primary": true} disconnected
> [07:57:34 ERR] MQTT {"primary": true} connection failed Connecting with MQTT server failed (NotAuthorized).: null
> ```

I saw exactly the same initially. The UI was loading but all devices were unavailable (which makes sense since companion was not connecting to mqtt). As soon as I added the mqtt details the UI is just a black page for me now

**Comment by 51av0sh** [comment]
Just wanted to report everything is working fine on my end now. I had optimization disabled when having the issue I reported above but I enabled optimization yesterday and everything started to work. No more 404s.

**Comment by DTTerastar** [comment]
I just tested:

[16:11:10 INF] Connecting to mqtt server at core-mosquitto:1883...
[16:11:11 INF] MQTT {"primary": true} connected

It seems to work just fine for me.

**Comment by DTTerastar** [comment]
[16:17:31 INF] MQTT {"primary": true} disconnected
[16:17:31 ERR] MQTT {"primary": true} connection failed Connecting with MQTT server failed (NotAuthorized).: null
[16:18:01 INF] MQTT {"primary": true} disconnected
[16:18:01 ERR] MQTT {"primary": true} connection failed Connecting with MQTT server failed (NotAuthorized).: null
[16:18:26 INF] HTTP GET /api/state/config responded 200 in 181.7499 ms
[16:18:27 INF] HTTP GET /api/state/devices responded 200 in 24.4517 ms
[16:18:28 INF] HTTP GET /api/state/devices responded 200 in 1.4829 ms
[16:18:31 INF] MQTT {"primary": true} disconnected
[16:18:31 ERR] MQTT {"primary": true} connection failed Connecting with MQTT server failed (NotAuthorized).: null

Yup, confirmed.

**Comment by DTTerastar** [comment]
Fixed in 0.6.2
